stable_baselines3/common/save_util.py

```diff
@@ -164,81 +168,198 @@ def json_to_data(json_string: str,
                 base64_object = base64.b64decode(serialization.encode())
                 deserialized_object = cloudpickle.loads(base64_object)
             except RuntimeError:
-                warnings.warn(f"Could not deserialize object {data_key}. " +
-                              "Consider using `custom_objects` argument to replace " +
-                              "this object.")
+                warnings.warn(
+                    f"Could not deserialize object {data_key}. "
+                    + "Consider using `custom_objects` argument to replace "
+                    + "this object."
+                )
             return_data[data_key] = deserialized_object
         else:
             # Read as it is
             return_data[data_key] = data_item
     return return_data
 
 
-def save_to_zip_file(save_path: str, data: Dict[str, Any] = None,
-                     params: Dict[str, Any] = None, tensors: Dict[str, Any] = None) -> None:
+@functools.singledispatch
+def open_path(path, mode, verbose=0, suffix=None):
+    """
+    Generic version of open_path. It verifies that the path is an io.BufferedIOBase, that the file is
+        writable if mode is "w", not writable if mode is
```

Comment:
Since this is the main entry function, I think all the functionality should be described here in like in the `path` parameter (e.g: "Open a file, creating necessary parent directories and renaming if files exist"). 

Also minor style thing: No extra indent in description on new lines.